ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/sequence/actions/FootstepPlanActionPlanningThread.java

```diff
@@ -123,13 +125,32 @@ else if (definition.getPlannerInitialStanceSide().getValue() == InitialStanceSid
 
       if (!isPreviewPlanner)
          state.getLogger().info("Planning footsteps...");
+
       FootstepPlannerOutput footstepPlannerOutput = footstepPlanner.handleRequest(footstepPlannerRequest, isPreviewPlanner);
-      FootstepPlan footstepPlan = footstepPlannerOutput == null ? new FootstepPlan() : footstepPlannerOutput.getFootstepPlan();
-      if (!isPreviewPlanner)
-         state.getLogger().info("Footstep planner completed with {}, {} step(s)", footstepPlannerOutput.getFootstepPlanningResult(), footstepPlan.getNumberOfSteps());
+      boolean plannerBusy = footstepPlannerOutput == null;
+      boolean foundSolution = !plannerBusy && footstepPlannerOutput.getFootstepPlanningResult() == FootstepPlanningResult.FOUND_SOLUTION;
+
+      while (plannerBusy) // Retry until planner is free
```

Comment:
We retry in the case the planner is already planning. The timeout is 5 s so shouldn't be an issue. However, we don't want to plan until successful. That should be part of a fallback node.